controllers/auth_config_controller.go

```diff
@@ -503,7 +503,14 @@ func (r *AuthConfigReconciler) translateAuthConfig(ctx context.Context, authConf
 				}
 			}
 
-			translatedAuthorization.KubernetesAuthz, err = authorization_evaluators.NewKubernetesAuthz(authorinoUser, authorization.KubernetesSubjectAccessReview.Groups, authorinoResourceAttributes)
+			var authorinoGroups expressions.Value
+			if authorization.KubernetesSubjectAccessReview.AuthorizationGroups != nil {
+				authorinoGroups, err = valueFrom(authorization.KubernetesSubjectAccessReview.AuthorizationGroups)
+				if err != nil {
+					return nil, err
+				}
+			}
+			translatedAuthorization.KubernetesAuthz, err = authorization_evaluators.NewKubernetesAuthz(authorinoUser, authorization.KubernetesSubjectAccessReview.Groups, authorinoGroups, authorinoResourceAttributes)
```

Comment:
I was thinking... What about resolving `groups` (if provided) or `authorizationGroups` (if provided) in the control plane?

Instead of passing both to the constructor, the constructor could expect always and only an `expressions.Value`. If `authorizationGroups != nil`, it's straightforward and we pass `valueFrom(authorizationGroups)`. Otherwise and if `len(groups) > 0`, we build a `json.JSONValue` (implements `expressions.Value`) setting by `Static` to `groups`.

Reply:
Thanks @guicassolato, I made the suggested change to accept a single groups arg.